src/Lumi/Components/Form.purs

```diff
@@ -764,6 +768,59 @@ arrayModal { label, addLabel, defaultValue, summary, component, componentProps }
       , validate: pure xs
       }
 
+editableTable
+  :: forall props row result
+   . { addLabel :: String
+     , defaultValue :: row
+     , form :: FormBuilder { readonly :: Boolean | props } row result
+     , labels :: Array String
+     , maxRows :: Int
+     , summary :: JSX
+     }
+  -> FormBuilder
+      { readonly :: Boolean
+      | props
+      }
+      (Array row)
+      (Array result)
+editableTable { addLabel, defaultValue, form, labels, maxRows, summary } =
+  FormBuilder \props@{ readonly } rows ->
+    let
+      fs = (un FormBuilder form) props <$> rows
+    in
+      { edit: \onChange -> pure $
+          Child
+            { key: Nothing
+            , child:
+                EditableTable.editableTable
+                  { addLabel
+                  , maxRows
+                  , readonly
+                  , rows: Left (Array.mapWithIndex Tuple rows)
+                  , rowEq: unsafeRefEq
+                  , summary
+                  , onRowAdd: onChange (flip Array.snoc defaultValue)
+                  , onRowRemove: \(Tuple i _) -> onChange \rows' -> fromMaybe rows' (Array.deleteAt i rows')
+                  , columns: labels # Array.mapWithIndex \j label ->
+                      { label
+                      , renderCell: \(Tuple i _) -> fold do
+                          { edit } <- Array.index fs i
+                          tree <- Array.index (edit \f -> onChange (\rows' -> fromMaybe rows' (Array.modifyAt i f rows'))) j
```

Comment:
I think what's going on here could probably do with some comments. Am I correct in thinking that we are requiring that the labels provided in the record argument above must match exactly (i.e. the same number, and in the same order) with the elements of the array returned by the form's `edit` function for this to work properly? If so, that seems a bit less than ideal.

Did you consider looking at the `label` fields of `Node` constructors occurring in the tree instead of requiring the user to supply a `labels` array? I suppose the fact that they're JSX rather than String complicates matters a bit though.

Reply:
Yes, exactly. I'll add some comments on this.

In a first pass I was using the `label` field of `Node` constructors, and changed `editableTable` to have column labels be `JSX` instead of `String`, though I didn't know the exact impact this could have. Perhaps this is a better solution for the moment? What do you think @spicydonuts?

I also just remembered that we need to render multiple form fields inside table cells side-by-side instead of one above the other. I will fix this.

Reply:
Changing the type to `JSX` seems fine to me. I'm not sure I understood the `label` bit -- you weren't talking about inspecting the `JSX` structure, right?

Reply:
Oh no, not at all. I meant just passing it around.

@hdgarrood another complicator for pulling `label` fields out of the `Tree` is that we should still be able to have `label` fields when the `value` is an empty array. Trying to extract `edit` from the supplied `FormBuilder` that way would return an empty `Forest` :/